charts/templates/statefulset.yaml

```diff
@@ -174,11 +174,10 @@ spec:
                         log "[poststart] wait for marklogic server to be ready"
                         sleep 5s
                       done
-                      sleep 10s
                       GROUP_CFG_TEMPLATE='{"group-name":"%s", "xdqp-ssl-enabled":"%s"}'
                       GROUP_CFG=$(printf "$GROUP_CFG_TEMPLATE" "$MARKLOGIC_GROUP" "$XDQP_SSL_ENABLED") 
                       log "Info: [poststart] Updating group configuration: ${GROUP_CFG}"
-                      curl --anyauth -m 20 -X PUT -H "Content-type: application/json" -d "${GROUP_CFG}" http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/groups/Default/properties --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}
+                      curl --retry 5 --retry-max-time 60 --anyauth -m 20 -s -X PUT -H "Content-type: application/json" -d "${GROUP_CFG}" http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/groups/Default/properties --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}
                       sleep 2s
```

Comment:
Why do we have still the sleep 2s ?

Reply:
Just noting that the answer should be a comment in the script, not in the PR - there can be legitimate uses for a `sleep` call, but they should always, always be documented so that engineers six months from now know why the sleep call is there.

Reply:
I think this sleep statement could be removed, as it seems to me that when the curl command returns, the group modification has been complete successfully. More detail could be find here: https://docs.marklogic.com/REST/PUT/manage/v2/groups/[id-or-name]/properties

Reply:
We should check the return code to see if the update caused a restart. If it did, the script should wait for the restart not have a sleep.

Reply:
> Just noting that the answer should be a comment in the script, not in the PR - there can be legitimate uses for a `sleep` call, but they should always, always be documented so that engineers six months from now know why the sleep call is there.

The task is to remove these sleeps, not document them.

Reply:
Removed sleep and added restart check based on return code from group update request.